lib/_emerge/depgraph.py

```diff
@@ -9157,6 +9162,49 @@ def display_problems(self):
 			show_mask_docs()
 			writemsg("\n", noiselevel=-1)
 
+		# Build a list of problematic packages
+		maintainer_nag_packages = []
+
+		for pkg in self._dynamic_config.digraph:
+			if not isinstance(pkg, Package):
+				continue
+
+			if not pkg.type_name == 'ebuild':
+				continue
+
+			maintainer_nag = pkg.root_config.settings.repositories[pkg.repo].maintainer_nag
+			maintainer_nag_url = pkg.root_config.settings.repositories[pkg.repo].maintainer_nag_url
+
+			if not maintainer_nag:
+				continue
+
+			path = os.path.dirname(pkg.root_config.trees['porttree'].dbapi.findname(pkg.cpv, myrepo=pkg.repo))
+			path = os.path.join(path, 'metadata.xml')
+
+			if MetaDataXML is not None and os.path.isfile(path):
+				try:
+					metadata_xml = MetaDataXML(path, None)
+					maintainer = metadata_xml.format_maintainer_string()
+				except SyntaxError:
+					continue
+
+				if not maintainer:
+					maintainer_nag_packages.append(pkg.cpv + "::" + pkg.repo)
+
+		if maintainer_nag_packages:
+			writemsg("\n" + colorize("WARN", "!!!") + \
```

Comment:
The first `!!!` is yellow, but not the other 2. Was that intentional? I think it might look better if they're all the same color.

Also, I'm a little concerned that this message might flood the terminal for a command like `emerge -e @world` for example. Maybe we should have some kind of limit on the number of packages displayed?

Reply:
Also, if maintainer_nag will be enabled by default, then we probably need the message to include instructions for how to disable it maintainer_nag.

Reply:
I suggest setting a flag, and displaying a message at the end with the list of packages attached.